rasterio/plot.py

```diff
@@ -105,31 +105,34 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
                 # will have the colors in the order expected by
                 # matplotlib (RGB)
                 arr = source.read(rgb_indexes, masked=True)
-                arr = reshape_as_image(arr)
-
             except KeyError:
-                arr = source.read(1, masked=True)
+                if indexes == None: indexes = (0,1,2)
+                indexes = tuple(i+1 for i in indexes)
```

Comment:
I don't think we need to add `+1` the users should pass valid indexes

Reply:
we also resolved this problem in our latest scripts.